src/mesh/Mesh.cpp

```diff
@@ -186,7 +186,7 @@ const PtrData &Mesh::data(const std::string &dataName) const
 PtrData &Mesh::createDataWithGradient(
```

Comment:
Couldn't we make the "withGradient" information an argument of `createData`?
Could make the overall logic a bit easier to understand.

Reply:
The problem with this option, is that we would have to set the spatial directions as well to a default value. An option would be to set the spatial dimensions by default to 1 for data when creating it (since this is irrelevant for data without gradient anyways) since we always use a vector to store all the values (even for vector data) but it not very "correct" in a sense that even data can be of vector values, but when creating the object this could be seen as a "wrong" input. I don't know in that case which is better. But this can be further discussed of course.